dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/SecondaryVoltageControl/Simplified/SecondaryVoltageControl.mo

```diff
@@ -79,40 +86,53 @@ equation
   frozen = FreezingActivated and ((blockedUp and (UpRefPu - UpPu) > 0) or (blockedDown and (UpRefPu - UpPu) < 0));
   switch1.u2 = frozen;
 
+  //level is sampled every tSample seconds to calculate levelDiscrete
+  if time >= tUpdate + tSample then
+    tUpdate = time;
+    levelDiscrete = level;
+  else
+    tUpdate = pre(tUpdate);
+    levelDiscrete = pre(levelDiscrete);
+  end if;
```

Comment:
This if condition could be changed to a when condition to avoid using pre() (As OpenModelica refuses the pre() with continuous variable). I tried to use the when condition, but could not build the model with Dynawo. From what I understood it is not yet implemented with discrete variables. Maybe this timeline could be updated in the future when it is implemented ?